### PR TITLE
[TCE] extract font styling to separate file

### PR DESCRIPTION
**Pull request by astronaut-wannabe**
## Why?
The last part of creating a turnkey solution for generating TCEs in additional languages is creating CSS for every new language. One of the things that make this difficult for someone who isn't familiar with the codebase is that all of the CSS is together, and you don't know what and where to look for doing the customizations.

## What?
 This creates the following structure:
```sh
website/app/assets/stylesheets/to_change_everything
.
├── _common.scss
├── _english.scss
├── _espanol.scss
├── _farsi.scss
├── _한국어.scss
├── common
│   ├── _font_styling.scss
│   ├── _navigation.scss
│   └── _reset.scss
├── farsi
│   └── _font_overrides.scss
└── korean
    └── _font_overrides.scss
```
The top-level files (e.g. `_farsi.scss`) are automatically generated. So are the sub-files (e.g. `farsi/_font_overrides.scss`). These are also automatically wired up and filled in with the same CSS as `font_styling.scss`.

A person generating a new TCE version then only has to look at 1 file (`farsi/_font_overrides.scss` in this example) and will only really need to delete any CSS that will be unchanged from the common styles and adjust some input values in the `scss` functions.

## Results 
Using this convention the custom css needed for the farsi or korean versions of TCE has been reduced to 50-60 lines of cookie-cutter CSS.

## note:
I do hope to do a similar thing to the layout code (margins, paddings, etc.) so that there will be a structure like this:
```sh
├── farsi
│   └── _font_overrides.scss
│   └── _layout_overrides.scss

```

### demos
- [To Change Everything: English][0]
- [To Change Everything: Espanol][1]
- [To Change Everything: فارسی][2]
- [To Change Everything: 한국어][3]

[0]:https://crimethinc-staging-pr-668.herokuapp.com/tce
[1]:https://crimethinc-staging-pr-668.herokuapp.com/tce/espanol
[2]:https://crimethinc-staging-pr-668.herokuapp.com/tce/فارسی
[3]:https://crimethinc-staging-pr-668.herokuapp.com/tce/한국어

### CODE REVIEW

**Comment by astronaut-wannabe** [comment]
@veganstraightedge is it possible to get a heroku deploy for this branch?

I want to check the layout in TOR and I don't think there is a way to do that locally

File: app/assets/stylesheets/to_change_everything/common/_font_styling.scss [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+@mixin pdfvideo-text($font-size, $line-height) { font: normal normal 600 $font-size /#{$line-height} $header-font-stack; }
+@mixin intro-header($font-size, $line-height)  { font: normal normal 500 $font-size /#{$line-height} $header-font-stack; }
+@mixin intro-text($font-size, $line-height)    { font: normal normal 600 $font-size /#{$line-height} $body-font-stack;   }
+@mixin imagetext($font-size, $line-height)     { font: normal normal 300 $font-size /#{$line-height} $header-font-stack  }
+@mixin body-text($font-size, $line-height)     { font: normal normal 500 $font-size /#{$line-height} $body-font-stack    }
+@mixin outro-text($font-size, $line-height)    { font: normal normal 600 $font-size /#{$line-height} $body-font-stack    }
+@mixin takeflight($font-size, $line-height)    { font: normal normal 800 $font-size /#{$line-height} $header-font-stack  }
+@mixin anarchy-text($font-size, $line-height)  { font: normal normal 600 $font-size /#{$line-height} $body-font-stack    }
+@mixin next-text($font-size, $line-height)     { font: normal normal 300 $font-size /#{$line-height} $header-font-stack  }
```
Comment (veganstraightedge) [inline app/assets/stylesheets/to_change_everything/common/_font_styling.scss]: Funny timing!

Just today at work, we ran into a weird ass issue with Heroku + Asset Pipeline + `font:` one-liners with `normal` in it.

The short and long of it is, don't one line `font` + `normal`. Weird things happen. It seems like Heroku's Asset Pipeline config is converting `normal` into `400`, but that causes unexpected behavior.

The work around was to do all of the `font-*` properties as their own rule.

File: app/assets/stylesheets/to_change_everything/common/_font_styling.scss [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+  text-indent: 22px;
+  color: white;
+  -webkit-font-smoothing: antialiased;
+  -moz-font-smoothing: antialiased;
+  -o-font-smoothing: antialiased;
+
+  @media (min-device-width:200px) and (max-width:600px) { @include anarchy-text(16px, 24px); }
+  @media (min-width:601px) and (max-width:1023px) {
+    @include anarchy-text(20px, 32px);
+    line-height: 32px; }
```
Comment (veganstraightedge) [inline app/assets/stylesheets/to_change_everything/common/_font_styling.scss]: I don't love this style with the closing `}` on the same line. It's shorter, yes. But the trade off is that editors can't easily auto-indent to reveal syntax / nesting issues.

File: app/assets/stylesheets/to_change_everything/common/_font_styling.scss [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+  @media (min-width:1024px) and (max-width:1200px) {
```
Comment (veganstraightedge) [inline app/assets/stylesheets/to_change_everything/common/_font_styling.scss]: A space after the colon is preferred.

Comment (astronaut-wannabe) [reply app/assets/stylesheets/to_change_everything/common/_font_styling.scss]: fixed

File: app/assets/stylesheets/to_change_everything/common/_font_styling.scss (hunk 2 above)
Comment (astronaut-wannabe) [reply app/assets/stylesheets/to_change_everything/common/_font_styling.scss]: fixed

**Comment by astronaut-wannabe** [comment]
@veganstraightedge did the following:
- multi-line CSS rules have `}` on new line
- Remove font-stack 1-liners
- add spaces after `:` in media queries

let me know if you spot anything else

**Pull request merged**